[Task1.9] Updating README file and UML diagrams per class feedback an…

selyard opened the pull request:
…d instructions.

[readme.md] Added significant details of project based on Phase 1 instructions

[docs/uml/activity_diagram.txt] Updated activity diagram with wheel speed calculation function.

[docs/uml/activity_diagram.png] Regenerated image.

[docs/uml/class_diagram.txt] Updated class diagram with new parameters (track, wheel size), functions (wheel speed calc).

[docs/uml/class_diagram.png] Regenerated image.

selyard committed:
[Task1.9] Updating README file and UML diagrams per class feedback and instructions.

[readme.md] Added significant details of project based on Phase 1 instructions

[docs/uml/activity_diagram.txt] Updated activity diagram with wheel speed calculation function.

[docs/uml/activity_diagram.png] Regenerated image.

[docs/uml/class_diagram.txt] Updated class diagram with new parameters (track, wheel size), functions (wheel speed calc).

[docs/uml/class_diagram.png] Regenerated image.

danielmohansahu commented on readme.md:
```diff
@@ -1,17 +1,51 @@
+## Personnel
+* Spencer Elyard, *TODO: Complete personnel blurb.*
+* Santosh Kesani, *TODO: Complete personnel blurb.*
+* Daneil Sahu, *TODO: Complete personnel blurb.*
```
```suggestion
* Daniel Sahu, *TODO: Complete personnel blurb.*
```

danielmohansahu commented on docs/uml/activity_diagram.txt:
```diff
@@ -46,6 +46,8 @@ Controller -> Test: throttle, steering
 Test -> MockPlant: command(throttle, steering)
 Test -> MockPlant: getState()
 MockPlant -> Test: speed, heading
+MockPlant -> Test: getWheelRPM()
```
I think this line is unnecessary; we're not using the resulting RPM value (that would be set prior to all this activity).

selyard replied:
I think we still need this calculation for the Phase 0 requirement to "output...the two drive wheel velocities". If we command a wheel RPM based on a centerline wheel, the actual left/right wheels will be +/- some RPM based on the different turning circle radii. Not an issue with a bicycle, but with a 4-wheel rover with individually driven wheels will need these corrections.

(EG: Rover with a steering angle sufficient to pivot around the right rear wheel. RR RPM will be 0, LR RPM will be some value, average of two would be centerline wheel RPM -> robot speed.)

danielmohansahu replied:
Hmm; I gotcha; I forgot about that requirement. In that case shouldn't it be:
```
Test -> Controller: getWheelRPM()
Controller -> Test: w_left, w_right
```
Or something like that? The `MockPlant` is just for our internal tests.

selyard replied:
Sounds good to me, I can update the txt and png here shortly.

selyard committed:
Update readme.md

Agreed - typo!

Co-authored-by: Daniel Sahu <danielmohansahu@gmail.com>

selyard committed:
[Task1.9] Updating README file and UML diagrams per class feedback and instructions.

[docs/uml/activity_diagram.txt] Updated activity diagram with correct placement for wheel speed getter function.

[docs/uml/activity_diagram.png] Regenerated image.
